yw_website/website/views.py

```
@@ -1,79 +1,73 @@
+        latest_version = Version.objects.filter(workflow=workflow).order_by('last_modified').first()
+        if latest_version is None:
+            workflow_list = workflow_list.exclude(pk=workflow.id)
+        else:
+            workflow.graph = latest_version.yw_graph_output
+            workflow.version_id = latest_version.id
+
+    paginator = Paginator(workflow_list, 10)
     page = request.GET.get('page')
-    documents = paginator.get_page(page)
+    workflows = paginator.get_page(page)
     host = request.get_host()
-    return render(request, 'pages/my_workflows.html', { 'document_list': documents, 'host': host })
 
-def detailed_workflow(request, document_id):
+    return render(request, 'pages/home_page.html', { 'workflow_list': workflows, 'host': host  })
+def detailed_workflow(request, workflow_id, version_id):
     try:
-        # TODO: change to get object or 404
         if request.method == "GET":
-            document = Workflow.objects.get(pk=document_id)
-            info = {'document': document}
+            form = request.POST
+            workflow = Workflow.objects.get(pk=workflow_id)
+            version = Version.objects.get(pk=version_id)
+            versions = Version.objects.filter(workflow=workflow)
+
+            runs = Run.objects.filter(version=version)
+            info = {'workflow': workflow, 'version': version, 'versions':versions ,'run_list':runs, 'form': form}
             return render(request, 'pages/detailed_workflow.html', info)
-    except Workflow.DoesNotExist:
+        elif request.method == "POST":
+            new_version = request.POST['version_id']
+            return redirect('/detailed_workflow/{}/{}/'.format(workflow_id, new_version))
+    
+    except ObjectDoesNotExist:
       return redirect(home)
 
-def run_detail(request):
-    document = Run.objects.get(id="1")
+def run_detail(request, run_id):
+    try:
+        document = Run.objects.get(pk=run_id)
+    except Run.DoesNotExist:
```

we probably want to throw a 404 error here